cmake/external/ngraph.cmake

```diff
@@ -0,0 +1,138 @@
+SET(NGRAPH_CODEGEN_LIB "${NGRAPH_INSTALL_DIR}/lib/libcodegen.so.0.8" CACHE FILEPATH "nGraph Codegen library." FORCE)
+SET(NGRAPH_CPU_LIB "${NGRAPH_INSTALL_DIR}/lib/libcpu_backend.so" CACHE FILEPATH "nGraph CPU backend library." FORCE)
+SET(NGRAPH_TBB_LIB "${NGRAPH_INSTALL_DIR}/lib/libtbb.so.2" CACHE FILEPATH "TBB library for nGraph." FORCE)
+MESSAGE(STATUS "Set ${NGRAPH_INSTALL_DIR}/lib to runtime path")
+SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_RPATH}" "${NGRAPH_INSTALL_DIR}/lib")
+
+INCLUDE_DIRECTORIES(${NGRAPH_INC_DIR}) # For nGraph code to include internal headers.
+INCLUDE_DIRECTORIES(${THIRD_PARTY_PATH}/install) # For Paddle code to include nGraph headers.
+
+SET(NGRAPH_GIT_REPO "https://github.com/NervanaSystems/ngraph.git")
+# SET(NGRAPH_GIT_TAG  "268853d0e7fbd0c87e36da506efb9505db0ef4dc")
+# SET(NGRAPH_GIT_TAG  "cce0c2249ca40e22a96fe91e46de76fb8ec4e95c")
+SET(NGRAPH_GIT_TAG  "b0b8719d7fd2dd5fce0ddc982fb57e876ddc0c00")
+
+SET(NGRAPH_USE_PREBUILT_LLVM FALSE)
+find_program(LSB_RELEASE_EXEC lsb_release)
+if(LSB_RELEASE_EXEC)
+    execute_process(COMMAND ${LSB_RELEASE_EXEC} -cs
+        OUTPUT_VARIABLE LSB_RELEASE_CODENAME
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+    string(COMPARE EQUAL "${LSB_RELEASE_CODENAME}" "xenial" XENIAL)
+    if(XENIAL)
+        SET(NGRAPH_USE_PREBUILT_LLVM TRUE)
+        message(STATUS "Configure nGraph with prebuilt LLVM for Ubuntu Xenial")
+    else()
+        message(STATUS "Configure nGraph with prebuilt LLVM for Ubuntu non-Xenial")
+    endif()
+else()
+    message(STATUS "Configure nGraph for non-Ubuntu")
+endif()
+
+IF(WITH_MKLDNN)
+    SET(NGRAPH_DEPENDS   ${MKLDNN_PROJECT})
+    SET(MKLDNN_INCLUDE_DIR    ${MKLDNN_INC_DIR})
+    SET(MKLDNN_LIB_DIR        ${MKLDNN_INSTALL_DIR}/lib)
+    MESSAGE(STATUS "Configure nGraph with MKLDNN: INCLDUE ${MKLDNN_INCLUDE_DIR}, LIB ${MKLDNN_LIB_DIR}")
+    ExternalProject_Add(
+        ${NGRAPH_PROJECT}
+        ${EXTERNAL_PROJECT_LOG_ARGS}
+        DEPENDS             ${NGRAPH_DEPENDS}
+        GIT_REPOSITORY      ${NGRAPH_GIT_REPO}
+        GIT_TAG             ${NGRAPH_GIT_TAG}
+        PREFIX              ${NGRAPH_SOURCES_DIR}
+        UPDATE_COMMAND      ""
+        CMAKE_ARGS          -DNGRAPH_INSTALL_PREFIX=${NGRAPH_INSTALL_DIR}
+        # CMAKE_ARGS          -DNGRAPH_DEX_ONLY=TRUE
+        CMAKE_ARGS          -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+        CMAKE_ARGS          -DNGRAPH_USE_PREBUILT_LLVM=${NGRAPH_USE_PREBUILT_LLVM}
+        CMAKE_ARGS          -DMKLDNN_INCLUDE_DIR=${MKLDNN_INCLUDE_DIR}
+        CMAKE_ARGS          -DMKLDNN_LIB_DIR=${MKLDNN_LIB_DIR}
+        CMAKE_CACHE_ARGS    -DNGRAPH_INSTALL_PREFIX:PATH=${NGRAPH_INSTALL_DIR}
+    )
+    ExternalProject_Add_Step(
+        ${NGRAPH_PROJECT}
+        PREDOWNLOAD
+        DEPENDERS download
+        COMMAND ${CMAKE_COMMAND} -E copy_directory ${MKLML_ROOT}/lib ${MKLDNN_LIB_DIR}
+        )
```

Comment:
since `SET(NGRAPH_DEPENDS   ${MKLDNN_PROJECT})`, you can remove line89-94

Reply:
will address in the cmake PR